modules/opengl/src/image/imagegl.cpp

```diff
@@ -82,18 +82,18 @@ void ImageGL::reAttachAllLayers(ImageType type) {
 void ImageGL::activateBuffer(ImageType type) {
     frameBufferObject_.activate();
 
-    std::vector<GLenum> drawBuffers{frameBufferObject_.getDrawBuffers()};
-    if (!drawBuffers.empty()) {
-        GLsizei numBuffersToDrawTo = static_cast<GLsizei>(drawBuffers.size());
+    drawBuffers_ = frameBufferObject_.getDrawBuffers();
```

Comment:
I don't fully understand why there is a need to cache these? They are in the frame buffer already?

Reply:
We modify them so we have to make a copy

Reply:
But why cache them when they are copied every time? They do not seem to be used elsewhere in imagegl?

Reply:
I want to avoid having to allocate that array every time... that new actually showed up on some of my profiles...

Reply:
Ok, the workaround should be documented because it is hard to know and could be removed later otherwise